src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPassBase.cs

```diff
@@ -285,6 +285,17 @@ protected static bool ShouldFormat(FormattingContext context, TextSpan mappingSp
                 return true;
             }
 
+            // Workaround for https://github.com/dotnet/aspnetcore/issues/36689
+            // A tags owner comes back as itself if it is preceeded by a HTML comment,
+            // because the whitespace between the comment and the tag is reported as not editable
+            if (owner is MarkupTextLiteralSyntax &&
```

Comment:
I like the first idea, but not sure about the second.

The question for me, is whether our testing is sufficient to ensure that these quirks don't break other scenarios, and don't get messy when you apply all of them at once, rather than targetted like is. If the answer to that is yes, then I would go further and create our own LocateOwner to use 100% of the time, and ban the compilers version.

Reply:
> The question for me, is whether our testing is sufficient to ensure that these quirks don't break other scenarios,

Ya that's a fair concern